main/src/ca/uwaterloo/flix/language/phase/optimizer/Inliner.scala

```diff
@@ -537,6 +537,46 @@ object Inliner {
     case _ => false
   }
 
+  /**
+    * Returns the definition of `sym` if it is let-bound and the [[shouldInlineVar]] predicate holds.
+    *
+    * Returns `default` otherwise.
+    *
+    * Throws an error if `sym` is not in scope. This also implies that it is the responsibility of the caller
+    * to replace any symbol occurrence with the corresponding fresh symbol in the variable substitution.
+    */
+  private def callSiteInline(sym: Symbol.VarSym, ctx0: LocalContext, default: => Expr)(implicit sym0: Symbol.DefnSym, root: MonoAst.Root, sctx: SharedContext, flix: Flix): Expr = {
```

Comment:
Why pass a => default function? We almost never use that pattern. Just return an option and let the caller deal with it.

Reply:
I did it to have `visitExp` in tail position in `callSiteIniline`